sdk.Dockerfile

```diff
@@ -0,0 +1,56 @@
+# Docker image with a build toolchain and environment variables set to use
+# the wasi-sdk sysroot. The SDK distribution must have first been built,
+# for example using docker_build.sh
+
+# Extract built SDK archive to copy out the sysroot
+FROM ubuntu:22.04 as dist
+
+ADD dist/wasi-sdk-*.*-linux.tar.gz /
+
+# Move versioned folder to unversioned to using bash glob to allow
+# this file to be independent of major version number.
+RUN mv /wasi-sdk-* /wasi-sdk
+
+# Use ubuntu to use official repository with newer cmake packages
+FROM ubuntu:22.04
+
+ENV LLVM_VERSION 15
+
+# Install build toolchain including clang, ld, make, autotools, ninja, and cmake
+RUN apt-get update && \
+    # Temporarily install to setup apt repositories
+    apt-get install -y curl gnupg && \
+\
+    curl -sS https://apt.llvm.org/llvm-snapshot.gpg.key | gpg --dearmor > /etc/apt/trusted.gpg.d/llvm.gpg && \
+    echo "deb [signed-by=/etc/apt/trusted.gpg.d/llvm.gpg] http://apt.llvm.org/jammy/ llvm-toolchain-jammy-${LLVM_VERSION} main" >> /etc/apt/sources.list.d/llvm.list && \
+    echo "deb-src [signed-by=/etc/apt/trusted.gpg.d/llvm.gpg] http://apt.llvm.org/jammy/ llvm-toolchain-jammy-${LLVM_VERSION} main" >> /etc/apt/sources.list.d/llvm.list && \
+\
+    curl -sS https://apt.kitware.com/keys/kitware-archive-latest.asc | gpg --dearmor > /etc/apt/trusted.gpg.d/kitware.gpg && \
+    echo "deb [signed-by=/etc/apt/trusted.gpg.d/kitware.gpg] https://apt.kitware.com/ubuntu/ jammy main" >> /etc/apt/sources.list.d/kitware.list && \
+    echo "deb-src [signed-by=/etc/apt/trusted.gpg.d/kitware.gpg] https://apt.kitware.com/ubuntu/ jammy main" >> /etc/apt/sources.list.d/kitware.list && \
```

Comment:
Surly the stock unbuntu version of cmake is good enough?

Reply:
Sure, was initially using debian where it was too old to support the `CMAKE_TOOLCHAIN_FILE` environment variable, but ubuntu's stock is fine